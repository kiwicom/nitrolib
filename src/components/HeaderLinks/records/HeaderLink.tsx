export type HeaderLink = {
  id: string,
  translation: string,
  newWindow: boolean,
  url: string,
};

export type SearchForm = {
  mode: string,
  destination: { type: string, name: string }, // TODO move proper search form into records @vacuum
  checkIn: Date,
  checkOut: Date | null,
  adults: number,
  children: number,
};

export type HeaderLinksContext =
  | "holidays"
  | "booking"
  | "kiwicom"
  | "pulkovoairport"
  | "kayakholidays";
