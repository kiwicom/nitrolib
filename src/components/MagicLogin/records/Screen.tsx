
export type Screen =
  | "intro"
  | "signUp"
  | "noAccount"
  | "kiwiLogin"
  | "facebookLogin"
  | "googleLogin"
  | "magicLink"
  | "signUpConfirmation"
  | "resetPassword"
  | "getSingleBooking";
