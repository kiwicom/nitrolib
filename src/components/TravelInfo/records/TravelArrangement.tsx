
export type Sector = {
  from: string,
  to: string,
  stops: "0" | "1" | "2" | "3",
  note: string,
};

export type TravelArrangement = {
  takeOff: Sector,
  landing: Sector,
};
