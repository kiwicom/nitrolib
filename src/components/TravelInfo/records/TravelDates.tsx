
export type TravelDates = {
  from: string,
  to: string,
};
