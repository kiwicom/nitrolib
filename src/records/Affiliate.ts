export type Affiliate = {
  id: string,
  params: { [key: string]: string },
};
