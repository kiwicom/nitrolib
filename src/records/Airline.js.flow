// @flow strict
export type Airline = {|
  id: string,
  lcc: number,
  name: string,
|};

export type Airlines = {|
  [key: string]: Airline,
|};

declare export var airlineDefault: Airline;
