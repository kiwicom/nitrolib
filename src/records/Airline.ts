export type Airline = {
  id: string,
  lcc: number,
  name: string,
};

export type Airlines = {
    [key: string]: Airline,
  };

// eslint-disable-next-line import/prefer-default-export
export const airlineDefault: Airline = { id: "xx", lcc: 0, name: "Default airline" };
