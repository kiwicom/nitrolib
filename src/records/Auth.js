// @flow strict
import type { User } from "./User";

export type AuthUser = {|
  type: "user",
  user: User,
  token: string,
|};

export type PasswordStrengthEnum = "WEAK" | "MEDIUM" | "STRONG";

export type AuthMagic = {|
  type: "magic",
  email: string,
  token: string,
|};

export type AuthToken = {|
  type: "token",
  bid: number,
  token: string,
|};

export type Auth = AuthUser | AuthMagic | AuthToken;

export type SocialProvider = "facebook" | "google";

export const getEmail = (a: Auth): string => {
  if (a.type === "user") {
    return a.user.email;
  }

  if (a.type === "magic") {
    return a.email;
  }

  return "";
};

// eslint-disable-next-line import/prefer-default-export
export const authDefault: Auth | null = null;
