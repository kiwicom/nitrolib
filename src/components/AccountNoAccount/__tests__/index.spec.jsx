// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import AccountNoAccount from "..";

import { brandDefault } from "../../../records/Brand";

describe("#AccountNoAccount", () => {
  test("renders button to register", () => {
    const wrapper = shallow(
      <AccountNoAccount
        onBack={() => {}}
        onRegister={() => {}}
        onFacebookLogin={() => {}}
        onGoogleLogin={() => {}}
      />,
    );
    const registerLink = shallow(<span>{wrapper.prop("children")(brandDefault)}</span>).find(
      `[t="account.register"]`,
    );

    expect(registerLink.exists()).toBe(true);
  });
});
