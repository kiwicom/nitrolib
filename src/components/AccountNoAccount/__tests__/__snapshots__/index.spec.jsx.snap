// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#AccountNoAccount render 1`] = `
<span>
  <ModalHeader>
    <Illustration
      name="NoResults"
      size="small"
    />
    <Heading
      element="h2"
    >
      <Translate
        html={false}
        t="account.no_bookings_or_account"
        transform={[Function]}
        values={Object {}}
      />
    </Heading>
    <Text
      t="account.no_bookings_or_account_description"
      values={
        Object {
          "brandName": "Kiwi.com",
        }
      }
    />
  </ModalHeader>
  <ModalSection>
    <Stack
      direction="column"
      spacing="natural"
    >
      <Button
        onClick={[Function]}
      >
        <Translate
          html={false}
          t="account.register"
          transform={[Function]}
          values={Object {}}
        />
      </Button>
      <ButtonLink
        iconLeft={<ChevronLeft />}
        onClick={[Function]}
        type="secondary"
      >
        <Translate
          html={false}
          t="account.back"
          transform={[Function]}
          values={Object {}}
        />
      </ButtonLink>
    </Stack>
  </ModalSection>
  <ModalSection
    suppressed={true}
  >
    <Text
      spaceAfter="medium"
      t="account.or_social_account"
      weight="bold"
    />
    <Stack
      flex={true}
      spacing="natural"
    >
      <Button
        block={true}
        bordered={true}
        icon={<Facebook />}
        onClick={[Function]}
        type="facebook"
      >
        <Translate
          html={false}
          t="account.log_in_with"
          transform={[Function]}
          values={
            Object {
              "provider": "Facebook",
            }
          }
        />
      </Button>
      <Button
        block={true}
        bordered={true}
        icon={<Google />}
        onClick={[Function]}
        type="google"
      >
        <Translate
          html={false}
          t="account.log_in_with"
          transform={[Function]}
          values={
            Object {
              "provider": "Google",
            }
          }
        />
      </Button>
    </Stack>
  </ModalSection>
</span>
`;
